### ISSUE TITLE
migrate to 1.20 -> invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by ashtonian** [rancher/terraform-provider-rancher2#753]
Running a deployment with the new provider seemed to cause a plugin error this is the tf logs. Let me what/know if you need more info. Cleared local tf files and tried again. 

```sh
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x27a0629]
goroutine 171 [running]:
github.com/terraform-providers/terraform-provider-rancher2/rancher2.expandClusterEKSConfigV2(0xc0003becb0, 0x1, 0x1, 0x4)
	/go/src/github.com/rancher/terraform-provider-rancher2/rancher2/structure_cluster_eks_config_v2.go:290 +0x369
github.com/terraform-providers/terraform-provider-rancher2/rancher2.expandCluster(0xc000edc770, 0xc000c24000, 0x0, 0x0)
	/go/src/github.com/rancher/terraform-provider-rancher2/rancher2/structure_cluster.go:510 +0x94a
github.com/terraform-providers/terraform-provider-rancher2/rancher2.resourceRancher2ClusterCreate(0xc000edc770, 0x2d0e160, 0xc0005fc180, 0x0, 0x0)
	/go/src/github.com/rancher/terraform-provider-rancher2/rancher2/resource_rancher2_cluster.go:123 +0xae
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0008861b0, 0xc000e56e10, 0xc0005e4a20, 0x2d0e160, 0xc0005fc180, 0x2a25801, 0xc0012641e8, 0xc00119b4a0)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/helper/schema/resource.go:310 +0x375
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000224300, 0xc001575a38, 0xc000e56e10, 0xc0005e4a20, 0xc001141b88, 0xc00159abb8, 0x2a28ae0)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0002d80d8, 0x307a090, 0xc001b878c0, 0xc000edc000, 0xc0002d80d8, 0xc001b878c0, 0xc000ed8ba0)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/helper/plugin/grpc_provider.go:885 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x2c45c60, 0xc0002d80d8, 0x307a090, 0xc001b878c0, 0xc0002dc180, 0x0, 0x307a090, 0xc001b878c0, 0xc000236000, 0x904)
	/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002a28c0, 0x30912d8, 0xc000580a80, 0xc000948000, 0xc0002dabd0, 0x3ebcf40, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:1210 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0002a28c0, 0x30912d8, 0xc000580a80, 0xc000948000, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:1533 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0001205b0, 0xc0002a28c0, 0x30912d8, 0xc000580a80, 0xc000948000)
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:871 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.34.0/server.go:869 +0x1fd

Error: The terraform-provider-rancher2_v1.20.0 plugin crashed!
```

**Comment by rawmind0** [comment]
Submitted PR https://github.com/rancher/terraform-provider-rancher2/pull/756 to fix the issue. The workaround would be specify `kubernetes_version` and `subnets` on the cluster definition

**Comment by rawmind0** [comment]
PR with fix merged. It will be included on next tfp release v1.20.1
